components/resolved_ts/src/endpoint.rs

```diff
@@ -101,129 +240,72 @@ impl ObserveRegion {
     }
 
     fn track_change_log(&mut self, change_logs: &[ChangeLog]) -> Result<()> {
-        match &mut self.resolver_status {
-            ResolverStatus::Pending {
-                locks,
-                tracked_index,
-                ..
-            } => {
-                for log in change_logs {
-                    match log {
-                        ChangeLog::Error(e) => {
-                            debug!(
-                                "skip change log error";
-                                "region" => self.meta.id,
-                                "error" => ?e,
-                            );
-                            continue;
-                        }
-                        ChangeLog::Admin(req_type) => {
-                            // TODO: for admin cmd that won't change the region meta like peer list
-                            // and key range (i.e. `CompactLog`, `ComputeHash`) we may not need to
-                            // return error
-                            return Err(box_err!(
-                                "region met admin command {:?} while initializing resolver",
-                                req_type
-                            ));
-                        }
-                        ChangeLog::Rows { rows, index } => {
-                            rows.iter().for_each(|row| match row {
-                                ChangeRow::Prewrite { key, start_ts, .. } => {
-                                    locks.push(PendingLock::Track {
-                                        key: key.clone(),
-                                        start_ts: *start_ts,
-                                    })
-                                }
-                                ChangeRow::Commit {
-                                    key,
-                                    start_ts,
-                                    commit_ts,
-                                    ..
-                                } => locks.push(PendingLock::Untrack {
-                                    key: key.clone(),
-                                    start_ts: *start_ts,
-                                    commit_ts: *commit_ts,
-                                }),
-                                // One pc command do not contains any lock, so just skip it
-                                ChangeRow::OnePc { .. } => {}
-                                ChangeRow::IngestSsT => {}
-                            });
-                            assert!(
-                                *tracked_index < *index,
-                                "region {}, tracked_index: {}, incoming index: {}",
-                                self.meta.id,
-                                *tracked_index,
-                                *index
+        if matches!(self.resolver_status, ResolverStatus::Pending { .. }) {
+            self.resolver_status
+                .track_change_log(self.meta.id, change_logs)
```

Comment:
Question: Why is this change needed?
It separates the implementation of ResolverStatus::Pending from other state, which may make the readability worse.